bindings/python/pymongocrypt/binary.py

```diff
@@ -76,19 +76,19 @@ def __init__(self):
 
 
 class MongoCryptBinaryIn(_MongoCryptBinary):
-    __slots__ = ("__copy",)
+    __slots__ = ("cref",)
 
     def __init__(self, data):
         """Creates a mongocrypt_binary_t from binary data."""
         # mongocrypt_binary_t does not own the data it is passed so we need to
-        # create a copy to keep the data alive.
-        self.__copy = ffi.new("uint8_t[]", data)
+        # create a separate reference to keep the data alive.
+        self.cref = ffi.from_buffer("uint8_t[]", data)
```

Comment:
This change does two things. 1) it allows both bytes and memoryview objects to be passed. The object returned by `from_buffer()` keeps a reference to `data`. 2) it removes the copy because from_buffer provides access to `data`'s internal buffer without copying.